### DISCUSSION TITLE
Does not work on OpenShift due to root permissions

### DISCUSSION THREAD

**Discussion by derFaizio** [gitpod-io/openvscode-server#195] [Q&A]
The image does not work out of the box on RedHat OpenShift Container Platform due to the root permissions (see error log below). Would like to request adaptations to the Dockerfile so the image is not run as a root user.
```
[93m[main 2021-09-29T14:36:07.848Z][0m rejected promise not handled within 1 second: Error: EACCES: permission denied, mkdir '/home/workspace/.opencode-server'
[93m[main 2021-09-29T14:36:07.849Z][0m stack trace: Error: EACCES: permission denied, mkdir '/home/workspace/.opencode-server'
[91m[main 2021-09-29T14:36:07.849Z][0m [Error: EACCES: permission denied, mkdir '/home/workspace/.opencode-server'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'mkdir',
  path: '/home/workspace/.opencode-server'
}
```

**Comment by piyoki** [comment]
Same issue for me.

**Comment by sluongng** [comment]
simple workaround is to create a Dockerfile like this

```
FROM gitpod/openvscode-server

USER root
```

Then build it and use it instead

**Comment by gclayburg** [comment]
I also get this error when running the docker command as a linux user that does not have a uid:gid of 1000:1000.  This image just assumes that the work directory mounted will be writable by the openvscode-server user.  In my case, I am using a groupid of 1001, so this Docker file works for me:

```
FROM gitpod/openvscode-server

USER root
RUN groupadd -g 1001 therealgroup
RUN usermod -G 1001 openvscode-server
USER openvscode-server
```

For this to work, you'll need to ensure the bind-mounted directory is group writable.
